homeassistant/components/squeezebox/media_player.py

```diff
@@ -203,24 +249,38 @@ def unique_id(self):
         """Return a unique ID."""
         return self._player.player_id
 
+    @property
+    def available(self):
+        """Return True if device connected to LMS server."""
+        return self._available
+
+    @available.setter
+    def available(self, val):
+        """Set available to True or False."""
+        self._available = bool(val)
+
     @property
     def state(self):
         """Return the state of the device."""
-        if self._player.power is not None and not self._player.power:
+        if not self.available:
+            return STATE_UNAVAILABLE
```

Comment:
We don't need to handle state unavailable. It's handled by the base entity class.